### PR TITLE
Update Cosmos docs

### LINKED ISSUE #1729: Update Cosmos documentation for 3.1
- Use async for EnsureCreated and EnsureDeleted
- Use partition key
  - Mention that `__partitionKey` should be used by default if creating the collection outside of EF
- Show how to remove the discriminator

### LINKED ISSUE #1771: Only works for SQL API
It should be specified that this provider only works for SQL API of Cosmos DB.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 738deb7b-d752-f628-3860-fe53891cadb0
* Version Independent ID: e0a18db9-d605-2a78-b77a-d22375dcfd4e
* Content: [Azure Cosmos DB Provider - EF Core](https://docs.microsoft.com/en-us/ef/core/providers/cosmos/)
* Content Source: [entity-framework/core/providers/cosmos/index.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/providers/cosmos/index.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @AndriySvyryd
* Microsoft Alias: **ansvyryd**

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Fixes #1729
Fixes #1771

### CODE REVIEW

File: entity-framework/core/modeling/data-seeding.md [hunk 1]
```diff
@@ -88,4 +88,4 @@ Depending on the constraints of your deployment the initialization code can be e
 * Running the initialization app locally
 * Deploying the initialization app with the main app, invoking the initialization routine and disabling or removing the initialization app.
 
-This can usually be automated by using [publish profiles](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/visual-studio-publish-profiles).
```
Comment (ajcvickers) [inline entity-framework/core/modeling/data-seeding.md]: Should we create AKA links for these?

File: entity-framework/core/providers/cosmos/limitations.md [hunk 2]
```diff
@@ -10,23 +10,23 @@ uid: core/providers/cosmos/limitations
 
 The Cosmos provider has a number of limitations. Many of these limitations are a result of limitations in the underlying Cosmos database engine and are not specific to EF. But most simply [haven't been implemented yet](https://github.com/aspnet/EntityFrameworkCore/issues?page=1&q=is%3Aissue+is%3Aopen+Cosmos+in%3Atitle+label%3Atype-enhancement+sort%3Areactions-%2B1-desc).
 
-## Temporary limitations
+## Temporary Limitations
```
Comment (ajcvickers) [inline entity-framework/core/providers/cosmos/limitations.md]: Why uppercase L?

File: entity-framework/core/modeling/data-seeding.md (hunk 1 above)
Comment (smitpatel) [reply entity-framework/core/modeling/data-seeding.md]: @ajcvickers - Is there specific reason we are creating AKA links? Content on docs site should be referenced through relative URLs (everything lives in docs.microsoft.com domain). The destination repos will take care of redirects and locales.

File: entity-framework/core/providers/cosmos/unstructured-data.md [hunk 3]
```diff
@@ -44,11 +43,11 @@ It is possible to access the properties that are not tracked by EF Core through
 
 ## Using CosmosClient
 
-To decouple completely from EF Core get the `CosmosClient` object that is [part of the Azure Cosmos DB SDK](https://docs.microsoft.com/en-us/azure/cosmos-db/sql-api-get-started) from `DbContext`:
+To decouple completely from EF Core get the `CosmosClient` object that is [part of the Azure Cosmos DB SDK](https://docs.microsoft.com/azure/cosmos-db/sql-api-get-started) from `DbContext`:
 
 [!code-csharp[CosmosClient](../../../../samples/core/Cosmos/UnstructuredData/Sample.cs?highlight=3&name=CosmosClient)]
 
-## Missing property values
+## Missing Property Values
```
Comment (ajcvickers) [inline entity-framework/core/providers/cosmos/unstructured-data.md]: I remember one of my Neuroscience processors telling me not to case sub-titles like this. Maybe he was wrong? Maybe thinks have changed? @bricelam? 🚎

File: entity-framework/core/modeling/data-seeding.md (hunk 1 above)
Comment (ajcvickers) [reply entity-framework/core/modeling/data-seeding.md]: @smitpatel Probably just me getting it wrong.

File: entity-framework/core/providers/cosmos/unstructured-data.md (hunk 3 above)
Comment (AndriySvyryd) [reply entity-framework/core/providers/cosmos/unstructured-data.md]: I'll revert.

**Pull request merged**